### ISSUE TITLE
Windows: Only create menu after windows are loading

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#21964]
No reason to call it before, we should call it as soon as the window starts loading to prevent waterfall on startup.

We can only do this on Windows because on other OS it is possible to interact with the menu before any window is opening. 

/cc @jrieken

**Comment by bpasero** [comment]
Turns out I cannot fix this. I tried everything:
* `Menu.setApplicationMenu(null)`
* Hiding the menu (can also be done simply by setting `"window.menuBarVisibility": "hidden"`)
* Not installing our menu (this will install electrons default menu)

=> I always end up with around `100ms` in `Menu` right after creating the Window. The time is spend it seems within the window creation, not even something I can control from the outside.

**Comment by jrieken** [comment]
>  I always end up with around 100ms in Menu right after creating the Window. The time is spend it seems within the window creation, not even something I can control from the outside.

Yes, it seems to be creating the window that populates the default menu. I have already moved our menu creation call [after](https://github.com/Microsoft/vscode/blob/master/src/vs/code/electron-main/main.ts#L266) the open call

**Comment by jrieken** [comment]
I have created https://github.com/electron/electron/issues/10947 on the Electron side

**Comment by Ramaqmar** [comment]
https://github.com/Microsoft/vscode/labels/perf#event-1337614830

**Comment by bpasero** [comment]
Closing as upstream.
